### ISSUE TITLE
Create S3FS sidecar to mount S3 folders and develop the Kubernetes spec files

### ISSUE THREAD

**Issue by gchang** [MAAP-Project/ZenHub#271]
Look at `https://github.com/efrecon/docker-s3fs-client` for one implementation. We can't use it verbatim as we have a different set of assumptions.
* We're using IAM roles for connecting to buckets
* We have different paths to the shared folders
* We're sharing the mount with another container. Note the `mountPropogation` flags needed to do so: https://github.com/signaleleven/s3fs-sidecar
* We also want the container to detect when the s3fs mount fails or dies and reconnect/remount as needed. There are probably several ways to accomplish this. We can have the s3fs sidecar container block on the s3fs command that's running in foreground mode. Or we could have the sidecar "watch" the mounted folders and kill/restart the s3fs process.

**Comment by gchang** [comment]
@bsatoriu If making a robust s3fs sidecar is going to take too much time, we can clean up the existing flaky implementation for now and use that for now. I prefer not having this block the installation of other things into GCC.

**Comment by bsatoriu** [comment]
The new sidecar repo is live at https://mas.maap-project.org/root/che-sidecar-s3fs. To dynamically mount the bucket to the right environment, I created a new [MAAP API environment endpoint](https://api.ops.maap-project.org/api/environment/config/test), and added a look up to this endpoint in the sidecar entrypoint script to fetch the environment based on the `CHE_API_EXTERNAL` env variable.  The new API endpoint references the same [environments.json ](https://github.com/MAAP-Project/maap-jupyter-ide/blob/develop/maap_environments.json) file that we're using for all the Jupyter plugins.

I also removed the MAAP_ENVIRONMENT ('ops') variable from the devfile registry definitions.
